tests/unit/db.js

```diff
@@ -39,6 +47,47 @@ function checkValueNotInDb(db, k, done) {
     });
 }
 
+function checkKeyInDB(db, key, cb) {
```

Comment:
It's confusing to have a check function that returns no error whether the object exists or not (if my understanding is correct): I suggest splitting into `checkKeyExistsInDB` and `checkKeyNotExistsInDB`, and using the appropriate one wherever it makes sense.